### PR TITLE
typings for defaultless settings

### PR DESCRIPTION
**Pull request by Vap0r1ze**
this adds real typings for settings definitions that dont have a `default` field:
- `COMPONENT` types are still always `any`
- `SELECT` types are `T | undefined` if there are no `{ default: boolean }` elements
- everything else is `T | undefined` if there is no default field

### CODE REVIEW

**Comment by Vap0r1ze** [comment]
@Captain8771 can you review as well

**Comment by Captain8771** [comment]
(i have no clue what is good practice in typing)

File: src/plugins/customRPC.tsx [hunk 1]
```diff
@@ -93,13 +93,13 @@ const numOpt = (description: string) => ({
...
 
-const choice = (label: string, value: any, _default?: Boolean) => ({
+const choice = (label: string, value: any, _default?: boolean) => ({
     label,
     value,
     default: _default
 }) as const;
 
-const choiceOpt = (description: string, options) => ({
+const choiceOpt = <T,>(description: string, options: T) => ({
```
Comment (Vendicated) [inline src/plugins/customRPC.tsx:102]: what the comma do

File: src/plugins/customRPC.tsx [hunk 2]
```diff
@@ -173,13 +173,13 @@ async function createActivity(): Promise<Activity | undefined> {
         activity.buttons = [
             buttonOneText,
             buttonTwoText
-        ].filter(Boolean);
+        ].filter((s): s is string => !!s);
```
Comment (Vendicated) [inline src/plugins/customRPC.tsx:176]: maybe can make this a util for reusability
```ts
function filterTruthy<T>(item: T): item is Exclude<T, 0 | "" | false | null | undefined> {
	return Boolean(s);
}

// usage
[...].filter(filterTruthy);
```

File: src/plugins/customRPC.tsx (hunk 1 above)
Comment (Vap0r1ze) [reply src/plugins/customRPC.tsx:102]: removes ambiguity in tsx files

File: src/plugins/customRPC.tsx (hunk 2 above)
Comment (Vap0r1ze) [reply src/plugins/customRPC.tsx:176]: REAL

File: src/plugins/customRPC.tsx (hunk 1 above)
Comment (Vendicated) [reply src/plugins/customRPC.tsx:102]: ah, i think just dont use arrow func and u wont need to

Comment (Vap0r1ze) [reply src/plugins/customRPC.tsx:102]: true but the return is just an object kinda cringe to

**Comment by Vap0r1ze** [comment]
yop

**Pull request merged**